modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/internal/exportimport/content/processor/test/DDMFormValuesExportImportContentProcessorTest.java

```diff
@@ -169,11 +170,20 @@ public void tearDown() throws Exception {
 		_journalArticleLocalService.deleteArticles(_liveGroup.getGroupId());
 
 		if (_ddmTemplate != null) {
-			_ddmTemplateLocalService.deleteDDMTemplate(_ddmTemplate);
+			_ddmTemplateLocalService.deleteTemplate(_ddmTemplate);
 		}
 
 		if (_ddmStructure != null) {
-			_ddmStructureLocalService.deleteDDMStructure(_ddmStructure);
+			boolean deleteInProcess = GroupThreadLocal.isDeleteInProcess();
+
+			try {
+				GroupThreadLocal.setDeleteInProcess(true);
```

Comment:
If we don't sent the `GroupThreadLocal.setDeleteInProcess` to true, following error:
```
com.liferay.dynamic.data.mapping.internal.exportimport.content.processor.test.DDMFormValuesExportImportContentProcessorTest > testReplaceJournalExportImportContentReferences FAILED
    com.liferay.dynamic.data.mapping.exception.RequiredStructureException$MustNotDeleteStructureReferencedByStructureLinks: Structure 44949 cannot be deleted because it is referenced by one or more structure links
        at com.liferay.dynamic.data.mapping.service.impl.DDMStructureLocalServiceImpl.deleteStructure(DDMStructureLocalServiceImpl.java:498)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at com.liferay.portal.spring.aop.AopMethodInvocationImpl.proceed(AopMethodInvocationImpl.java:50)
        at com.liferay.portal.spring.transaction.TransactionInterceptor.invoke(TransactionInterceptor.java:69)
        at com.liferay.portal.spring.aop.AopMethodInvocationImpl.proceed(AopMethodInvocationImpl.java:57)
        at com.liferay.portal.kernel.aop.ChainableMethodAdvice.invoke(ChainableMethodAdvice.java:64)
        at com.liferay.portal.spring.aop.AopMethodInvocationImpl.proceed(AopMethodInvocationImpl.java:57)
        at com.liferay.portal.kernel.aop.ChainableMethodAdvice.invoke(ChainableMethodAdvice.java:64)
        at com.liferay.portal.spring.aop.AopMethodInvocationImpl.proceed(AopMethodInvocationImpl.java:57)
        at com.liferay.portal.spring.aop.AopInvocationHandler.invoke(AopInvocationHandler.java:49)
        at com.sun.proxy.$Proxy939.deleteStructure(Unknown Source)
        at com.liferay.dynamic.data.mapping.internal.exportimport.content.processor.test.DDMFormValuesExportImportContentProcessorTest.tearDown(DDMFormValuesExportImportContentProcessorTest.java:182)
```

is thrown here: https://github.com/liferay-forms/liferay-portal/blob/22aaeac33ac6e10183c1fe8bd4aded4a98c822a0/modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/service/impl/DDMStructureLocalServiceImpl.java#L491-L499

Reply:
Hi @rodrigopaulino 

The objects are deleted in the Dataguard logic, after the test is finished.

Nevertheless, I have changed the code to delete both DLFileEntry and DLFileEntryType, this will avoid setting and unsetting the GroupThreadLocal.setDeleteInProcess flag, see new commit https://github.com/liferay-forms/liferay-portal/pull/1524/commits/cc7de0d614d4de11815ebe41a43f1589095d1265